src/main/java/de/tum/in/www1/exerciseapp/service/StatisticService.java

```diff
@@ -209,7 +211,8 @@ private void addResultToAllStatistics(QuizExercise quizExercise, Result result)
         for (Question question : quizExercise.getQuestions()) {
             // update QuestionStatistics with the result
             if (result != null && question.getQuestionStatistic() != null) {
-                question.getQuestionStatistic().addResult(((QuizSubmission) result.getSubmission()).getSubmittedAnswerForQuestion(question), result.isRated());
+                QuizSubmission quizSubmission = quizSubmissionRepository.findOne(result.getSubmission().getId());
```

Comment:
I think line 214 can also be moved before the for loop to improve the performance, yes?